solvebio/query.py

```diff
@@ -1220,3 +1220,26 @@ def fields(self):
             fields = [f for f in fields if f not in self._exclude_fields]
 
         return fields
+
+    def __len__(self):
+        base_len = super(QueryFile, self).__len__()
+        if base_len == float('inf'):
+            raise TypeError(
+                'Unable to determine the number of records for the '
+                'current file-based query'
+            )
+        return base_len
+
+    def __repr__(self):
+        try:
+            super(QueryFile, self).__repr__()
+        except TypeError:
+            return 'Query returned 0 results.'
+        else:
+            return super(QueryFile, self).__repr__()
+
+    def __next__(self):
```

Comment:
need to modify the superclass `next()` method instead of this...

From the jira ticket:
> if len(self) raises TypeError, don’t raise StopIteration since we are not yet at the end

In `next()` as it is currently defined, you will see this:

```
# len(self) returns `min(limit, total)` results
if not _is_join and self._cursor == len(self):
    raise StopIteration
```

With the new code, `len(self)` might raise TypeError.. so handle that case here but just continue normally if it does return TypeError...

```
try:
    if not _is_join and self._cursor == len(self):
        raise StopIteration
except TypeError
    # len(self) is unknown so just continue normally
    continue
```